### PR TITLE
Add JOSS publication

### PR DESCRIPTION
**Pull request by lgeiger**
This is a first draft of our submission to [The Journal of Open Source Software](https://joss.theoj.org). The aim is for a paper that gives a quick overview of the software package within 250-1000 words.
I'd be very happy to get feedback into which direction we should go with this paper and where we need to evaluate.

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - name: Plumerai Research
+    index: 1
+date: 9 September 2019
+bibliography: paper.bib
+---
+
+# Summary
+
+Society can be transformed by utilizing the power of deep learning outside of data centers: self-driving cars, mobile-based neural networks, smart edge devices, and autonomous drones all have the potential to revolutionize everyday lives.
+However, existing neural networks that use 32 bits, 16 bits or 8 bits to encode each weight and activation have an energy budget which is far beyond the scope for many of these applications.
```
Comment (timdebruin) [inline paper/paper.md]:
```suggestion
However, existing neural networks that use 32 bits, 16 bits or 8 bits to encode each weight and activation have an energy budget which is far beyond the scope of many of these applications.
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+Binarized Neural Networks (BNNs) have emerged as a promising solution to this problem. In these networks both weights and activations are restricted to $\{−1, +1\}$, resulting in models which are dramatically less computationally expensive, have a far lower memory footprint, and when executed on specialized hardware yield in a stunning reduction in energy consumption [@Courbariaux2016].
```
Comment (timdebruin) [inline paper/paper.md]:
```suggestion
Binarized Neural Networks (BNNs) have emerged as a promising solution to this problem. In these networks both weights and activations are restricted to $\{−1, +1\}$, resulting in models which are dramatically less computationally expensive, have a far lower memory footprint, and when executed on specialized hardware yield a stunning reduction in energy consumption [@Courbariaux2016].
```

**Comment by lgeiger** [comment]
> I would maybe emphasize the key advantages a bit more.

Great idea! @koenhelwegen What do you think about f3c38e0d5d61a63f1750492103070533b39bda95

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+`larq` is built around the concept of quantized layers that compute
+
+$$
+\sigma(f(q_{\, \mathrm{kernel}}(\boldsymbol{w}), q_{\, \mathrm{input}}(\boldsymbol{x})) + b)
+$$
+
+with full precision weights $\boldsymbol{w}$, arbitrary precision input $\boldsymbol{x}$, layer operation $f$ (e.g. $f(\boldsymbol{w}, \boldsymbol{x}) = \boldsymbol{x}^T \boldsymbol{w}$ for a densely-connected layer), activation $\sigma$ and bias $b$.
+$q_{\, \mathrm{kernel}}$ and $q_{\, \mathrm{input}}$ are quantizers that define an operation for quantizing a kernel and inputs, respectively, and a pseudo-gradient used for automatic differentiation.
+
+The flexible yet easy to use API of `larq` is aimed at researchers in the field of efficient deep learning [@2019arXiv190602107H] as well as at beginners.
```
Comment (koenhelwegen) [inline paper/paper.md]:
```suggestion
The flexible yet easy to use API of `larq` is aimed at researchers in the field of efficient deep learning [@2019arXiv190602107H] as well as at practitioners who want to explore BNNs for their application of interest. Furthermore, `larq` makes it much easier for beginners and students to get started with BNNs.
```

**Pull request merged**